### PR TITLE
T180653 - Add ip support to user search input

### PR DESCRIPTION
**Pull request by dayllanmaza**

### CODE REVIEW

File: src/components/fields/select-users.js [hunk 1]
```diff
@@ -36,22 +37,27 @@ class SelectUsers extends React.Component {
...
-					crossDomain: true
-				} )
-					.map( ( ajaxResponse ) => {
-						return ajaxResponse.response.query.globalallusers.map( ( user ) => {
-							return {
-								label: user.name,
-								value: user.name
-							};
-						} );
+				if ( isIPAddress( input ) ) {
```
Comment (Niharika29) [inline src/components/fields/select-users.js]: These changes could use some comments to explain what exactly is happening here.

Comment (dayllanmaza) [reply src/components/fields/select-users.js]: I thought it was self explanatory but I can add a line if you think it is necessary

**Review by kaldari — COMMENTED** [review]
Any chance we could get some tests for isIPAddress()? MediaWiki uses QUnit for JS testing, FWIW.

**Comment by dayllanmaza** [comment]
Welp, I thought of that, but we don't have any test framework setup right now. Maybe we should do that as part of a different task? (Setting up a test framework and get code coverage to a decent %)

**Review by kaldari — APPROVED** [review]
Looks reasonable, although I haven't tested all these complicated regexes. Let's create a new task for adding a test framework.

**Comment by dayllanmaza** [comment]
Great. I've created this one to keep track of it.
https://phabricator.wikimedia.org/T183988

**Pull request merged**